src/connection.rs

```diff
@@ -34,100 +44,244 @@ impl<C: Connector> EppConnection<C> {
             connector,
             greeting: String::new(),
             timeout,
-            ready: false,
+            current: None,
+            next: None,
         };
 
-        this.greeting = this.get_epp_response().await?;
-        this.ready = true;
+        this.read_greeting().await?;
         Ok(this)
     }
 
-    /// Constructs an EPP XML request in the required form and sends it to the server
-    async fn send_epp_request(&mut self, content: &str) -> Result<(), Error> {
-        let len = content.len();
+    async fn read_greeting(&mut self) -> Result<(), Error> {
+        assert!(self.current.is_none());
```

Comment:
Do we want to throw a more informative error here?

Reply:
`read_greeting()` is a private method on `EppConnection`, it has two uses, for both of which it's pretty trivial to check that this condition holds: I don't think so.